http.go

```diff
@@ -176,9 +176,16 @@ func (z *Zoho) HTTPRequest(endpoint *Endpoint) (err error) {
 	dataType := reflect.TypeOf(endpoint.ResponseData).Elem()
 	data := reflect.New(dataType).Interface()
 
-	err = json.Unmarshal(body, data)
-	if err != nil {
-		return fmt.Errorf("Failed to unmarshal data from response for %s: got status %s: %s", endpoint.Name, resolveStatus(resp), err)
+	if len(body) > 0 { // Avoid failed to unmarshal if there is no result
+		err = json.Unmarshal(body, data)
+		if err != nil {
+			return fmt.Errorf("Failed to unmarshal data from response for %s: got status %s: %s", endpoint.Name, resolveStatus(resp), err)
+		}
+
+		// Search for hidden errors (appears on success response)
+		if bytes.Contains(body, []byte(`"status":"error"`)) {
+			return fmt.Errorf("%s", string(body))
+		}
```

Comment:
Im not opposed, but I do wonder if there might be unintended consequences on other Zoho APIs that have been implemented here.

I've been working on a testing branch but haven't had the guts to _test_ yet. Hopefully it will make decisions like this easier to make.

Reply:
It shouldn't, it's a simple if. I came with this solution because there were no errors reported because the API is still returning SUCCESS, but the record is not inserted/updated.

also, i think is a better way to change `if len(body) > 2` because when is empty it comes with `[]`